### PR TITLE
[NPM-3156] Rewrite TestReadInitialState to use os assigned ports

### PR DESCRIPTION
**Pull request by hmahmood**
### What does this PR do?

Rewrite `TestReadInitialState[TCP|UDP]` to use os assigned ports. Also adds checking for ports listening in the root network namespace in addition to a non-root one.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/port_test.go [hunk 1]
```diff
@@ -52,141 +50,136 @@ func TestMain(m *testing.M) {
...
+		t.Cleanup(func() { cmd.Process.Kill() })
+		if port != 0 {
+			return nil
 		}
-		for _, p := range ports[2:] {
-			if _, ok := initialPorts[PortMapping{nsIno, p}]; !ok {
-				t.Errorf("PortMapping(test ns) returned false for port %d", p)
-				return false
+
+		scanner := bufio.NewScanner(stderr)
```
Comment (akarpz) [inline pkg/network/port_test.go:84]: so if we pass a `0` port to socat it returns the port it assigned to `sterr`?

Comment (hmahmood) [reply pkg/network/port_test.go:84]: In its logs, yes.

File: pkg/network/port_test.go [hunk 2]
```diff
@@ -52,141 +50,136 @@ func TestMain(m *testing.M) {
...
+	require.NoError(t, err)
+	t.Cleanup(func() { ns.Close() })
+	rootNs, err = kernel.GetRootNetNamespace("/proc")
+	require.NoError(t, err)
+	t.Cleanup(func() { rootNs.Close() })
+
+	var protos []string
+	switch proto {
+	case "tcp":
+		protos = []string{"tcp4", "tcp6", "udp4", "udp6"}
```
Comment (akarpz) [inline pkg/network/port_test.go:127]: why do we still spin up udp servers when running in tcp mode and vice-versa? Do we do this on `main`?

File: pkg/network/port_test.go (hunk 1 above)
Comment (akarpz) [reply pkg/network/port_test.go:84]: that is weird and interesting to me

File: pkg/network/port_test.go (hunk 2 above)
Comment (hmahmood) [reply pkg/network/port_test.go:127]: Yes, we spin up both udp and tcp to test if just listing tcp or udp doesn't include the other protocol ports.

**Review by leeavital — APPROVED** [review]

**Pull request merged**